src/App.js

```diff
@@ -1,12 +1,219 @@
-import React from 'react';
+import React, {useState, useEffect} from 'react';
+import axios from "axios";
+import PokemonPage from "./PokemonPage";
 import './App.css';
+import PokemonList from "./PokemonList";
+import PokemonLogo from "./assets/img.png"
 
 function App() {
-  return (
-    <div>
-      Begin hier met de opdracht!
-    </div>
-  );
+    const [pokemon, setPokemon] = useState([]);
+    const [error, setError] = useState('');
+    const [loading, toggleLoading] = useState(false);
+    const [nextPageUrl, setNextPageUrl] = useState('');
+    const [previousPageUrl, setPreviousPageUrl] = useState('');
+    const [currentPageUrl, setCurrentPageUrl] = useState(`https://pokeapi.co/api/v2/pokemon/?offset=0&limit=20`);
+
+    function goToNextPage() {
+        setCurrentPageUrl(nextPageUrl);
+    }
+
+    function goToPreviousPage() {
+        setCurrentPageUrl(previousPageUrl);
+    }
+
+    useEffect(() => {
+        setError('');
+        toggleLoading(true);
+
+        async function pokemon() {
+            try {
+                const {data} = await axios.get(currentPageUrl)
+                setPokemon(data.results);
+                setNextPageUrl(data.next);
+                setPreviousPageUrl(data.previous);
+
+            } catch (e) {
+                setError("Er is iets mis gegeaan!")
+                console.error(e)
+            }
+            toggleLoading(false);
+        }
+
+        pokemon();
+
+    }, [currentPageUrl]);
+
+
+    return (
+        <>
+            <img className='logo' src={PokemonLogo} alt='PokemonLogo'/>
+
+            <div className="pokemonCard">
+                {error && <p>{error}</p>}
+                {loading && <p>Pagina wordt geladen!</p>}
+
+                {pokemon && pokemon.map((pokemon) => {
+
+                    return <PokemonList key={pokemon.name} name={pokemon.name} currentPageUrl={currentPageUrl}/>
+                })}
+            </div>
+
+            <PokemonPage
+                goToNextPage={nextPageUrl ? goToNextPage : null}
```

Comment:
nextPageUrl is van zichzelf al `null` als je een keer `Volgende` en `Vorige` klikt. Daarom kun je gewoon `goToNextPage={nextPageUrl}` doen.